tests/test_apply.py

```diff
@@ -51,11 +51,46 @@
     qml.T
 ]
 
+test_transpile_options = [
+    {},
+    {'optimization_level':2},
+    {'optimization_level':2, 'seed_transpiler':22}
+]
+
+test_device_options = [
+    {},
+    {'optimization_level':3},
+    {'optimization_level':1}
+]
+
 single_qubit_operations_param = [qml.PhaseShift, qml.RX, qml.RY, qml.RZ]
 two_qubit = [qml.CNOT, qml.SWAP, qml.CZ]
 two_qubit_param = [qml.CRZ]
 three_qubit = [qml.Toffoli, qml.CSWAP]
 
+@pytest.mark.parametrize("qiskit_device", [AerDevice, BasicAerDevice])
```

Comment:
Would suggest using the `device` fixture that is defined in the `tests/conftest.py` file (it includes the parametrizing of devices). Other test cases include examples for its usage. The `@pytest.mark.parametrize("analytic", [True])` mark might need to be defined for the class in such a case.